### PR TITLE
[13.0] Migrate sale_exception

### PR DESCRIPTION
**Pull request by simahawk**
Take over #1051 to speed it up. 

Done here:

* fix version
* fix linting
* fix base test class
* removed legacy override of `_exception_rule_eval_context`

:warning:  WARNING: if you are already using this in custom exceptions, replace `sale` and `sale_line` with `obj` in your code snippets.

### CODE REVIEW

File: sale_exception/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @api.model
+    def _reverse_field(self):
+        return "sale_ids"
+
+    def _detect_exceptions(self, rule):
+        records = super(SaleOrderLine, self)._detect_exceptions(rule)
+        return records.mapped("order_id")
+
+    @api.model
+    def _exception_rule_eval_context(self, rec):
```
Comment (simahawk) [inline sale_exception/models/sale_order_line.py]: TODO: shall we drop this then? Checking...

Comment (simahawk) [reply sale_exception/models/sale_order_line.py]: removed!

**Review by TDu — APPROVED** [review]
Thank you for tackling this

**Review by jgrandguillaume — APPROVED** [review]
LGTM

**Pull request merged**